src/ledger/test/LedgerTxnTests.cpp

```diff
@@ -2804,6 +2819,308 @@ TEST_CASE("Erase performance benchmark", "[!hide][erasebench]")
+            }
+            REQUIRE(successfulFailures <= expectedFailedKeys.size());
+        };
+
+    auto runTest = [&](size_t cacheSize) {
+        Config cfg = getTestConfig();
+        cfg.ENTRY_CACHE_SIZE = cacheSize;
+
+        // Test setup.
+        VirtualClock clock;
+        cfg.EXPERIMENTAL_BUCKETLIST_DB = true;
+        Application::pointer app = createTestApplication(clock, cfg);
+        UnorderedSet<LedgerKey> keysToPrefetch;
+        auto& root = app->getLedgerTxnRoot();
+        LedgerTxn ltx(root);
+
+        UnorderedSet<LedgerKey> expectedSuccessKeys;
+        UnorderedSet<LedgerKey> expectedFailedKeys;
+        std::set<LedgerEntry> entrySet;
+        std::set<LedgerKey> deadKeySet;
+
+        auto lkMeterCold = std::make_shared<LedgerKeyMeter>();
+        auto lkMeterHot = std::make_shared<LedgerKeyMeter>();
+
+        auto entries =
+            LedgerTestUtils::generateValidUniqueLedgerEntriesWithExclusions(
+                {TTL}, 3);
+
+        auto deadKeys =
+            LedgerTestUtils::generateValidUniqueLedgerEntryKeysWithExclusions(
+                {CONFIG_SETTING}, 1);
+        auto ttlEntry = LedgerTestUtils::generateValidLedgerEntryOfType(TTL);
+        entries.push_back(ttlEntry);
+        for (auto e : entries)
+        {
+            e.lastModifiedLedgerSeq = 1;
+            ltx.createWithoutLoading(e);
+            entrySet.emplace(e);
+        };
+        for (const auto& k : deadKeys)
+        {
+            ltx.eraseWithoutLoading(k);
+            deadKeySet.emplace(k);
+        }
+
+        // Insert all entries into the database.
+        std::vector<LedgerEntry> ledgerVect{entrySet.begin(), entrySet.end()};
+        std::vector<LedgerKey> deadKeyVect{deadKeySet.begin(),
+                                           deadKeySet.end()};
+
+        app->getBucketManager().addBatch(*app, 2, 20, {}, ledgerVect,
+                                         deadKeyVect);
+        ltx.commit();
+
+        auto addTxn = [&](int32_t offset, std::vector<LedgerEntry> entries,
+                          std::vector<LedgerKey> deadKeys = {}) {
+            SorobanResources resources;
+
+            for (auto& e : entries)
+            {
+                resources.readBytes += xdr::xdr_size(e);
+                auto k = LedgerEntryKey(e);
+                keysToPrefetch.emplace(k);
+                // Randomly add the key to either the read or write set.
+                if (stellar::rand_flip())
+                {
+                    resources.footprint.readOnly.emplace_back(k);
+                }
+                else
+                {
+                    resources.footprint.readWrite.emplace_back(k);
+                }
+                // If the key is not a TTL and its offset is less than 0 and
+                // it has not been marked as expected to succeed in another
+                // transaction, add it to the expected failed keys set.
+                if (k.type() != TTL && offset < 0 &&
```

Comment:
I think it's kinda complicated to manage `expectedSuccessKeys` and `expectedFailedKeys` in this lambda. It's also not really testing the feature properly, as `expectedFailedKeys` can all load and the test still passes. I think `addTxn` should just setup the footprint and lkMeter object, and we can pass the set of expected failed/succeeded keys manually to `checkPrefetch`.